Fixed package.json for root, client and server.

Ambient004 opened the pull request:
Some development dependencies were being mixed with dependencies that were supposed to be used at run-time. These have been fixed leading to smaller production sizes. Developers should remember to `npm install` for the new changes to apply and optionally `npm prune` to clear unused Node packages. When using `npm install` make sure to use `--save-dev` or `-D` when appropriate.

Laudenum committed:
Fixed package.json for root, client and server.

Some development dependencies were being mixed with dependencies that
were supposed to be used at runtime. These have been fixed. Developers
should remember to `npm install` for the new changes to apply and
optionally `npm prune` to clear unusued Node packages.

kvoli committed:
add group page, group index, remove & add user group (#68)

* Change upload API to return a https url to the image

* Fix bug causing ArtifactGrid to render vertically instead of horizontally

* fix console errors

error 1: unique key people, error 2: forwardRef issue

* big diff, add 7 new shallow render tests

* update travis.yml to include client testing

* remove redundant imports into file testing >>> to pass linting

* ..lol added groups

kvoli commented:
Travis seems to be unable to find React-Scripts

kvoli reviewed (changes requested):
Fix React-Scripts dependency so test's can run on Travis

Laudenum committed:
Merge commit 'fd23a5c72c77c0462fd367f74ccdf66cf3a5997f' into ambient004

Laudenum committed:
Updated packages and fixed react-scripts.

TravisCI fails when react-scripts is added in devDepdencies.
Moving it back to dependencies should fix it.

Laudenum committed:
Moved all client package.json devDependencies.

kvoli reviewed (approved):
Excellent!
